File: problems/problems_3024/solution.go
package problem3024

import (
	"encoding/json"
	"log"
	"strings"
)

func triangleType(nums []int) string {
	a, b, c := nums[0], nums[1], nums[2]
	mn, mx := min(a, b, c), max(a, b, c)
	remain := a + b + c - mx - mn
	if mx >= mn+remain {
		return "none"
	}
	if a == b && a == c {
		return "equilateral"
	}
	if a == b || a == c || b == c {
		return "isosceles"
	}
	return "scalene"
}

func Solve(inputJsonValues string) interface{} {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}

	return triangleType(nums)
}

File: problems/problems_3024/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.triangleType(test_input)

    def triangleType(self, nums: List[int]) -> str:
        if nums[0] + nums[1] <= nums[2] or nums[2] + nums[1] <= nums[0] or nums[0] + nums[2] <= nums[1]:
            return "none"
        if nums[0] == nums[1] == nums[2]:
            return "equilateral"
        if nums[0] == nums[1] or nums[0] == nums[2] or nums[1] == nums[2]:
            return "isosceles"
        return "scalene"
